src/main/java/edu/iu/terracotta/controller/app/OutcomeController.java

```diff
@@ -0,0 +1,210 @@
+            if(outcomeDto.getOutcomeId() != null){
+                log.error(TextConstants.ID_IN_POST_ERROR);
+                return new ResponseEntity(TextConstants.ID_IN_POST_ERROR, HttpStatus.CONFLICT);
+            }
+
+            outcomeDto.setExposureId(exposureId);
+            Outcome outcome;
+            try{
+                outcome = outcomeService.fromDto(outcomeDto);
+            } catch (DataServiceException ex) {
+                return new ResponseEntity("Unable to create Outcome: " + ex.getMessage(), HttpStatus.BAD_REQUEST);
+            }
+
+            Outcome outcomeSaved = outcomeService.save(outcome);
+            OutcomeDto returnedDto = outcomeService.toDto(outcomeSaved, false);
+
+            HttpHeaders headers = new HttpHeaders();
+            headers.setLocation(ucBuilder.path("/api/experiments/{experiment_id}/exposures/{exposure_id}/outcomes/{outcome_id}")
+                    .buildAndExpand(outcome.getExposure().getExperiment().getExperimentId(), outcome.getExposure().getExposureId(), outcome.getOutcomeId()).toUri());
+            return new ResponseEntity<>(returnedDto, headers, HttpStatus.CREATED);
+        } else {
+            return new ResponseEntity(TextConstants.NOT_ENOUGH_PERMISSIONS, HttpStatus.UNAUTHORIZED);
+        }
+    }
+
+
+    @RequestMapping(value= "/{experiment_id}/exposures/{exposure_id}/outcomes/{outcome_id}", method = RequestMethod.PUT)
+    public ResponseEntity<Void> updateOutcome(@PathVariable("experiment_id") Long experimentId,
+                                              @PathVariable("exposure_id") Long exposureId,
+                                              @PathVariable("outcome_id") Long outcomeId,
+                                              @RequestBody OutcomeDto outcomeDto,
+                                              HttpServletRequest req)
+            throws ExperimentNotMatchingException, OutcomeNotMatchingException, BadTokenException {
+
+        log.info("Updating outcome with id {}", outcomeId);
+        SecurityInfo securityInfo = apijwtService.extractValues(req, false);
+        apijwtService.experimentAllowed(securityInfo, experimentId);
+        apijwtService.outcomeAllowed(securityInfo, experimentId, exposureId, outcomeId);
+
+        if(apijwtService.isInstructorOrHigher(securityInfo)){
+            Optional<Outcome> outcomeSearchResult = outcomeService.findById(outcomeId);
+
+            if(!outcomeSearchResult.isPresent()){
+                log.error("Unable to update. Outcome with id {} not found.", outcomeId);
+                return new ResponseEntity("Unable to update. Outcome with id " + outcomeId + TextConstants.NOT_FOUND_SUFFIX, HttpStatus.NOT_FOUND);
+            }
+            Outcome outcomeToChange = outcomeSearchResult.get();
+            outcomeToChange.setTitle(outcomeDto.getTitle());
+            outcomeToChange.setMaxPoints(outcomeDto.getMaxPoints());
+            outcomeToChange.setExternal(outcomeDto.getExternal());
```

Comment:
If an outcome is changed from external to internal, then we surely want to set the type to NONE and the LmsOutcomeId to null.